### ISSUE TITLE
[BUG][BATCH_NORM] avoid persistent pipeline hang under tilelang 0.1.9

### ISSUE THREAD

**Issue by RMLYC** [tile-ai/TileOPs#1263]
# [BUG][BATCH_NORM] avoid persistent pipeline hang under tilelang 0.1.9

### Symptom / Motivation

Issue #1074 tracks remaining TileLang 0.1.9 skip failures. One failure class is a BatchNorm persistent pipeline hang: `BatchNormFwdOp` / `BatchNormBwdOp` persistent kernels launch and then hang at CUDA synchronization when the persistent path uses `num_stages=1`.

The affected TileOps code path is the persistent BatchNorm forward/backward path in `tileops/kernels/norm/batch_norm.py`. In the current persistent config, `block_l = L`, so the pipelined loop has exactly one tile:

```python
for l_tile in T.Pipelined(L // block_l, num_stages=num_stages):
    T.copy(..., x_shared)
```

### Root Cause Analysis

The generated TileLang 0.1.9 CUDA for BatchNorm forward persistent `num_stages=1` is missing the required arrival on the forward data barrier. The generated producer path contains:

```cpp
mbarrier[0].expect_transaction(128);
tl::tma_load(..., mbarrier[0], 128);
```

The consumer path waits on the same barrier:

```cpp
mbarrier[0].wait(0);
```

However, `mbarrier[0]` is initialized with an arrival count of one:

```cpp
mbarrier[0].init(1);
```

`expect_transaction(128)` only accounts for the TMA transaction bytes; it does not perform a barrier arrival. Since the generated CUDA has no `mbarrier[0].arrive()` or `mbarrier[0].arrive_and_expect_tx(...)`, the consumer wait can never observe the required arrival and the kernel hangs.

TileLang 0.1.8 generated `arrive_and_expect_tx(...)` for the same stage-1 persistent forward path:

```cpp
mbarrier[0].arrive_and_expect_tx(128);
tl::fence_proxy_async();
tl::tma_load(..., mbarrier[0], 128);
```

This supplies both the arrival and the expected TMA transaction bytes, so the consumer wait can complete.

For TileOps, the practical issue is that the BatchNorm persistent path is a single-tile path (`block_l == L`). There is no next tile to prefetch, so the stage-1 producer/consumer pipeline has little opportunity for load/compute overlap but introduces TMA/mbarrier synchronization risk.

### Related Files

- `tileops/kernels/norm/batch_norm.py`
- `tests/ops/test_batch_norm.py`
- `tests/ops/test_norm_ops.py`

## Goal

Unblock the BatchNorm persistent-path failures tracked by #1074 by making TileOps avoid the TileLang 0.1.9 stage-1 persistent pipeline hang while preserving BatchNorm correctness.

## Plan

1. Change BatchNorm forward persistent default and autotune configs in `tileops/kernels/norm/batch_norm.py` from `num_stages=1` to `num_stages=0`.
2. Apply the same `num_stages=0` persistent config change to BatchNorm backward.
3. Keep non-persistent BatchNorm configs unchanged; they already use `num_stages=0`.
4. Remove only the BatchNorm-specific TileLang 0.1.9 skips once the exact skipped tests pass.
5. Verify the BatchNorm torch.compile smoke tests and `tests/ops/test_batch_norm.py -m smoke`

## Constraints

- Do not broaden skip coverage.
- Do not remove skips for unrelated TileLang 0.1.9 failure classes from #1074.
- Keep the change scoped to the TileOps BatchNorm persistent path.
- Preserve the existing BatchNorm public API and tensor semantics.

## Acceptance Criteria

- [ ] BatchNorm forward persistent configs no longer select `num_stages=1`.
- [ ] BatchNorm backward persistent configs no longer select `num_stages=1`.
- [ ] `tests/ops/test_norm_ops.py::TestBatchNormCustomOp::test_fwd_torch_compile_smoke` passes.
- [ ] `tests/ops/test_norm_ops.py::TestBatchNormCustomOp::test_bwd_torch_compile_smoke` passes.
- [ ] `tests/ops/test_batch_norm.py -m smoke` passes.
- [ ] BatchNorm-specific skips are removed without affecting unrelated #1074 failure classes.
